awscrt/aio/aiohttp.py

```diff
@@ -0,0 +1,436 @@
+
+        Returns:
+            bytes: The next chunk of data from the response body.
+                Returns empty bytes when the stream is completed and no more chunks are left.
+        """
+        if self._received_chunks:
+            return self._received_chunks.popleft()
+        elif self._completion_future.done():
+            return b""
+        else:
+            future = Future()
+            self._chunk_futures.append(future)
+            return await asyncio.wrap_future(future, loop=self._loop)
+
+    async def wait_for_completion(self) -> int:
+        """Wait asynchronously for the stream to complete.
+
+        Returns:
+            int: The response status code.
+        """
+        return await self._completion_future
+
+    async def _set_request_body_generator(self, body_iterator: AsyncIterator[bytes]):
+        ...
+
+
+class AIOHttpClientStream(AIOHttpClientStreamUnified):
+    """Async HTTP stream that sends a request and receives a response.
+
+    Create an AIOHttpClientStream with `AIOHttpClientConnection.request()`.
+
+    Attributes:
+        connection (AIOHttpClientConnection): This stream's connection.
+
+        completion_future (asyncio.Future): Future that will contain
+            the response status code (int) when the request/response exchange
+            completes. If the exchange fails to complete, the Future will
+            contain an exception indicating why it failed.
+
+    Notes:
+        All async method on a stream (await stream.next(), etc.) must be performed in the
+        thread that owns the event loop used to create the stream
+    """
+
+    def __init__(self, connection: AIOHttpClientConnection, request: HttpRequest,
+                 loop: Optional[asyncio.AbstractEventLoop] = None) -> None:
+        """Initialize an HTTP client stream.
+
+        Args:
+            connection (AIOHttpClientConnection): The connection to send the request on.
+            request (HttpRequest): The HTTP request to send.
+            loop (Optional[asyncio.AbstractEventLoop]): Event loop to use for async operations.
+                If None, the current event loop is used.
+        """
+        super().__init__(connection, request, loop=loop)
+
+
+class AIOHttp2ClientStream(AIOHttpClientStreamUnified):
+    """HTTP/2 stream that sends a request and receives a response.
+
+    Create an AIOHttp2ClientStream with `AIOHttp2ClientConnection.request()`.
+    """
+
+    def __init__(self,
+                 connection: AIOHttpClientConnection,
+                 request: HttpRequest,
+                 request_body_generator: AsyncIterator[bytes] = None,
+                 loop: Optional[asyncio.AbstractEventLoop] = None) -> None:
+        super().__init__(connection, request, request_body_generator=request_body_generator, loop=loop)
+
+    async def _write_data(self, body, end_stream):
+        future: Future = Future()
+        body_stream: InputStream = InputStream.wrap(body, allow_none=True)
```

Comment:
This isn't a blocker, but you don't need to annotate types on assignment unless the type you want to declare is wider than the type being assigned. For example:

```python
optional_message: str | None = None
if self.message_available():
    optional_message = self.read_message()
```

Another example of when you'd do this if you want to widen parametric types on a generic.